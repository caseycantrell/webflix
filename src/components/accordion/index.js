import { useState } from "react";

export default function Accordion({ children, ...restProps }){
    return (
        <Container {...restProps}>
            <Inner>
                {children}
            </Inner>
        </Container>
    )
};

Accordion.Title = function AccordionTitle({ children, ...restProps }) {
    return <Title {...restProps}></Title>
};

Accordion.Frame = function AccordionFrame({ children, ...restProps }) {
    return <Frame {...restProps}></Frame>
};

Accordion.Item = function AccordionItem({ children, ...restProps }) {
    const [toggleShow, setToggleShow] = useState(false);
    return <Item {...restProps}></Item>
};

Accordion.Header = function AccordionHeader({ children, ...restProps }) {
    return <Header {...restProps}></Header>
};