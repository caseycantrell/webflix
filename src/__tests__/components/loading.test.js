import { render } from "@testing-library/react";
import { Loading } from "../../components";

describe("<Loading />", () => {
    it("renders the <Loading /> component", () => {
        const { container, getByTestId } = render(
            <Loading src="/images/casey.png" data-testid="loading" />
        );

        expect(getByTestId("loading")).toBeTruthy();
        expect(getByTestId("loading-picture")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();

    });

    it("renders the <Loading.ReleaseBody /> component", () => {
        const { container } = render(
            <Loading.ReleaseBody data-testid="loading" />
            );

        expect(container.firstChild).toMatchSnapshot();
    })
});