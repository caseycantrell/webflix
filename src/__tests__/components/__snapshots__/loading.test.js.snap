// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders the <Loading /> component 1`] = `
<div
  class="sc-fWCJzd igszUd"
  data-testid="loading"
>
  <img
    class="sc-dvQaRk joObTn"
    data-testid="loading-picture"
    src="/images/users//images/casey.png.png"
  />
</div>
`;

exports[`<Loading /> renders the <Loading.ReleaseBody /> component 1`] = `null`;
